core/src/avm2/domain.rs

```diff
@@ -219,6 +219,15 @@ impl<'gc> Domain<'gc> {
         res
     }
 
+    pub fn get_defined_names(&self) -> Vec<QName<'gc>> {
```

Comment:
This could return an `impl Iterator<Item = QName<'gc>>` to avoid the need to allocate a `Vec`

Reply:
I had tried that, but the compiler told me that a type annotation was needed for the `collect` call and to consider specifying `Vec`. And when I tried to skip the `collect`, the compiler told me that I was capturing a lifetime that "does not appear in bounds", which I believe makes sense? So I settled for specifying `Vec`, despite agreeing that the allocation is unfortunate. Would you have more advice towards this? I feel my lack of rust experience is showing

Reply:
@friedkeenan I didn't realize that there was an intermediate `read()` call. It's probably more trouble that it's worth to make an `impl Iterator` work here.